### ISSUE TITLE
osqueryd and recursive glob in mid-path

### ISSUE THREAD

**Issue by thomasareed** [facebook/osquery#3213]
I'm unclear whether this is a bug or simply something that should be clarified in the docs.

On macOS, using osqueryd and file integrity monitoring with a path like the following has odd behavior:

`/Users/%%/foo/bar/%`

I expected that this should match, and log changes within, paths like the following:

`/Users/joe/foo/bar/`
`/Users/joe/a/b/c/foo/bar/`

However, in testing, it seems that this logs every change in any user folder, as if it were matching folders like these:

`/Users/joe/blah/`
`/Users/bill/`
`/Users/bob/a/b/c/`

In other words, it will match paths as if everything following the `%%` were discarded.

If this is invalid syntax, and the `%%` can only appear at the end of a path, unlike a single `%`, then that should be noted in the file integrity monitoring docs.

On the other hand, if this syntax should actually match any path beginning with `/Users/` and ending with `/foo/bar/`, then this looks like a bug on macOS (at least).

**Comment by jedi22** [comment]
Yes, when I designed the system it was only ever intended for %% to be used at the end of globbing paths. Since then the code has been updated but it seems this restriction has stayed. I will make sure the docs are updated.

**Comment by thomasareed** [comment]
Sounds good, thanks!
